worker/src/handlers/download.rs

```diff
@@ -27,17 +28,12 @@ fn prepare_request(url: &str, range_start: u64, range_end: u64) -> reqwest::Requ
 fn calculate_next_even_second(current: DateTime<Utc>) -> DateTime<Utc> {
     let millis = current.timestamp_millis() % 1000;
     let remaining_millis = 1000 - millis;
-    let remaining_millis_u64 = if remaining_millis > 0 {
-        remaining_millis as u64
-    } else {
-        0
-    };
 
-    current + Duration::from_millis(remaining_millis_u64)
+    current + Duration::milliseconds(remaining_millis)
 }
 
 /// Sleep until the start time of the job
-async fn wait_for_start_time(payload: &JobMessage) -> Result<()> {
+async fn wait_for_start_time(payload: &JobMessage) -> Result<(), anyhow::Error> {
```

Comment:
Any reason to add error type explicitly here? Should work just fine with `Result<()>`